arch/arm/soc/st_stm32/stm32f4/soc_pinmux.c

```diff
@@ -15,32 +15,63 @@
 #define PAD(AF, func)				\
 				[AF - 1] = func
 
+#define PINMUX_PWM(port, chan, pin)		\
+	PAD(STM32F4_PINMUX_FUNC_##pin##_PWM##port##_CH##chan,\
+	    STM32F4X_PIN_CONFIG_AF_PUSH_UP),
+
+#define PINMUX_UART(port, dir, pin)		\
+	PAD(STM32F4_PINMUX_FUNC_##pin##_##port##_##dir, \
+	    STM32F4X_PIN_CONFIG_AF_PUSH_UP),
+
+/* Blank pinmux by default */
+#define PINMUX_PWM2(pin, chan)
+#define PINMUX_UART1(dir, pin)
+#define PINMUX_UART2(dir, pin)
+
+#ifdef CONFIG_PWM_STM32_2
+	#undef PINMUX_PWM2
+	#define PINMUX_PWM2(pin, chan)		PINMUX_PWM(2, chan, pin)
+#endif
+
+#ifdef CONFIG_UART_STM32_PORT_1
+	#undef PINMUX_UART1
+	#define PINMUX_UART1(dir, pin)		PINMUX_UART(USART1, dir, pin)
+#endif
+
+#ifdef CONFIG_UART_STM32_PORT_2
+	#undef PINMUX_UART2
+	#define PINMUX_UART2(dir, pin)		PINMUX_UART(USART2, dir, pin)
+#endif
+
+#define PINMUX_UART_TX(pin, port)		PINMUX_UART##port(TX, pin)
+#define PINMUX_UART_RX(pin, port)		PINMUX_UART##port(RX, pin)
+
 static const stm32_pin_func_t pin_pa0_funcs[] = {
```

Comment:
Yes it would be good to extend to other STM32 as I also said in the cover letter. This can be done as a follow-up if you agree, especially considering that I do not have other STM32 on my desk right now.

Sure, I will look into the consistency issue, thanks.